project_app/components/Dices.js

```diff
@@ -0,0 +1,113 @@
+import styled from "styled-components";
+import {useEffect, useState} from 'react';
+
+
+
+export function DiceRating({rating, setRating}) {
+
+    const [viewRating, setViewRating] = useState(0);
+
+    useEffect(() => {
+        setViewRating(rating);

```

Comment:
hmm im worried about the rerenders in this component. you are setting parent state that controls this component, then sets local state based on parent state.